### PR TITLE
api request returns stackflow questions' title based on tags

### LINKED ISSUE #5: `?EXAMPLE` returns something from the Stack Overflow/Exchange API
This doesn't need to be the right method yet, just so we can prove that we can query the API and get something back in the console. We can make it work later for the actual method.

Story points: 5

### PR DESCRIPTION
**Pull request by krixstin**
tags are 

- "api" by default
- "method" from user 
- "library" from user

### CODE REVIEW

File: src/api_stackflow.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//install npm i node-fetch --save
+
+const fetch = require("node-fetch"); 
+
+var method=""
+var library=""
```
Comment (dcarpenter31) [inline src/api_stackflow.js]:
```suggestion
let method="";
let library="";
```

Comment (dcarpenter31) [reply src/api_stackflow.js]: we should use `let` or `const` for variables when we can. `var` is a bit outdated as all variables created with it are global

File: src/api_stackflow.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//install npm i node-fetch --save
```
Comment (dcarpenter31) [inline src/api_stackflow.js]: could you run `npm i node-fetch` and push the updated `package.json`? we'll all be able to install the new dependency automatically with `npm i` once that file is updated

File: src/api_stackflow.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    console.log("\n" , "METHOD: ", method.toUpperCase() ,"\n", "LIBRARY:", library.toUpperCase(), "\n")
+    
+    getapi(api_url);
+    
+    async function getapi(url){ 
+        
+        const response =await fetch(url); //store fetch data in "response"
+        var data = await response.json(); //store "response" in form of Json
+
+        // console.log(data); //see json file
```
Comment (dcarpenter31) [inline src/api_stackflow.js]:
```suggestion
```

File: src/api_stackflow.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+        // it will print top 5 Q's title
+        for (let i = 0; i < 5; i++){
+            var title = data.items[i].title;
```
Comment (dcarpenter31) [inline src/api_stackflow.js]:
```suggestion
            let title = data.items[i].title;
```

File: src/api_stackflow.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            console.log("Title",i+1 ,": ", title)
+        }
+    }
+}
+
+get_stackflow("post","java");
```
Comment (dcarpenter31) [inline src/api_stackflow.js]:
```suggestion
export default get_stackflow
```

Comment (dcarpenter31) [reply src/api_stackflow.js]: this will allow us to import the function in say, `index.ts`.

**Review by dcarpenter31 — CHANGES_REQUESTED** [review]

File: src/api_stackflow.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    //data is sorted based on "votes"
+    api_url="https://api.stackexchange.com/2.2/questions?order=asc&sort=votes&tagged=api%3B" + method + "%3B" + library + "&site=stackoverflow"   
+    console.log("\n" , "METHOD: ", method.toUpperCase() ,"\n", "LIBRARY:", library.toUpperCase(), "\n")
+    
+    getapi(api_url);
+    
+    async function getapi(url){ 
+        
+        const response =await fetch(url); //store fetch data in "response"
+        var data = await response.json(); //store "response" in form of Json
```
Comment (dcarpenter31) [inline src/api_stackflow.js]:
```suggestion
        let data = await response.json(); //store "response" in form of Json
```

**Comment by dcarpenter31** [comment]
Closes #5

File: src/api_stackflow.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+const fetch = require("node-fetch"); 
+
+let method="";
+let library="";
```
Comment (dcarpenter31) [inline src/api_stackflow.js]:
```suggestion
```

Comment (dcarpenter31) [reply src/api_stackflow.js]: no need to declare these since they are declared as function parameters below

**Review by dcarpenter31 — CHANGES_REQUESTED** [review]
I tried calling `get_stackflow` with the arguments "print" and "java" and got the following result:

```
METHOD:  PRINT 
 LIBRARY: JAVA 

Title 1 :  API for printers?
Title 2 :  How do you print off a Java Book?
Title 3 :  how to force Java print graphics in 300dpi
Title 4 :  Print Microsoft Office and PDF files in Java
/home/electricolor/Documents/s2021/modsim/ExampleBot/build/src/api_stackflow.js:56
                            title = data.items[i].title;
                                                  ^

TypeError: Cannot read property 'title' of undefined
    at /home/electricolor/Documents/s2021/modsim/ExampleBot/build/src/api_stackflow.js:56:51
    at step (/home/electricolor/Documents/s2021/modsim/ExampleBot/build/src/api_stackflow.js:33:23)
    at Object.next (/home/electricolor/Documents/s2021/modsim/ExampleBot/build/src/api_stackflow.js:14:53)
    at fulfilled (/home/electricolor/Documents/s2021/modsim/ExampleBot/build/src/api_stackflow.js:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by krixstin** [comment]
@carpenterd777 I figured that there were only 4 posts that have "API", "print" and "java" tags from my API request. 

I reformatted the function from "returning something" from StackOverflow to 
"returning one post link of a question that's answered & returning error message if none of reqs met"

**Review by dcarpenter31 — CHANGES_REQUESTED** [review]
lets try and have the bot return the actual results from one or more stack exchange posts. if we return a link, the user would be able to get the same functionality from google instead, and the idea of our bot was to be accessible from within discord

**Review by dcarpenter31 — APPROVED** [review]

**Pull request merged**